GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs

```diff
@@ -49,24 +49,22 @@ public static void LoadGameOnline()
         }
     }
 
-    public static ServerResponseModel verifyAndSaveUserPurchase(string purchaseToken)
+    public static void verifyAndSaveUserPurchase(Product product)
     {
         var values = new Dictionary<string, string>
         {
-            ["purchaseToken"] = purchaseToken
+            ["receipt"] = getReceiptDetails(product.receipt),
+            ["isSubscription"] = product.definition.id.Contains("subscription").ToString()
         };
 
-        return sendUnityWebRequest(values, VERIFY_AND_SAVE_TOKEN_URL);
+        ServerResponseModel serverResponse = sendUnityWebRequest(values, VERIFY_AND_SAVE_TOKEN_URL);
+        PurchaseController.ConfirmPendingPurchase(product, serverResponse.success);
     }
 
-    public static ServerResponseModel verifyPurchaseWithApi(string packageName, string productId, string purchaseToken)
+    private static string getReceiptDetails(string receipt)
     {
-        var values = new Dictionary<string, string>
-        {
-            ["packageName"] = packageName,
-            ["productId"] = productId,
-            ["purchaseToken"] = purchaseToken
-        };
-        return sendUnityWebRequest(values, VERIFY_PURCHASE_WITH_API_URL);
+        receipt = String.Join("", receipt.Split('\\'));
+        receipt = receipt.Remove(0, receipt.IndexOf("orderId") - 2);
```

Comment:
Don't make change to it at client. What if you feel you need extra info? Just send all info to server.

Reply:
I see 👍